tests/python/pants_test/engine/test_rules.py

```diff
@@ -201,6 +202,32 @@ def goal_rule_returning_a_non_goal() -> int:
         return 0
 
 
+def fmt_task_func(func: Callable) -> str:
+  """Generate the str for a Rust Func, which is how Rust refers to `@rule`s."""
+  return f"{func.__module__}:{func.__code__.co_firstlineno}:{func.__name__}"
```

Comment:
Since this method is at least being used by one other test file, might it be useful to put in testutil/etc?

Reply:
I wasn't sure what to do here. Do we expect files other than `test_rules.py`, `test_scheduler.py`, and `test_engine.py` to use rule graph validation output? (Genuine question, I'm not sure).

If so, certainly this should move to `testutil.engine.util`. If not, I'm not sure what to do?